events/oio_events_queue_beanstalkd.c

```diff
@@ -478,3 +484,19 @@ _q_is_stalled (struct oio_events_queue_s *self)
 		return FALSE;
 	return ((guint)l) >= oio_events_common_max_pending;
 }
+
+static gint64
+_q_get_health(struct oio_events_queue_s *self)
+{
+	struct _queue_BEANSTALKD_s *q = (struct _queue_BEANSTALKD_s*) self;
+	EXTRA_ASSERT(q != NULL && q->vtable == &vtable_BEANSTALKD);
+
+	gint64 limit = G_MAXINT64;
+	if (oio_events_beanstalkd_check_level_deny > 0)
+		limit = oio_events_beanstalkd_check_level_deny;
+	else if (oio_events_beanstalkd_check_level_alert > 0)
+		limit = oio_events_beanstalkd_check_level_alert;
+
+	// TODO: take g_async_queue_length (q->queue) into account
+	return 100 - (q->pending_events * 100 / limit);
```

Comment:
I propose to reserve this piece of logic for the case of an explicitly configured hard limit with `oio_events_beanstalkd_check_level_deny`.

But when no configuration has been set, the limit at G_MAXINT64 will ~always give a 100% health and be unusable to compare the service using their score. We could use a function smoothly decreasing, e.g. `1 + 1/ln(x+1)`.